GitCommands/RevisionReader.cs

```diff
@@ -227,75 +235,68 @@ private bool TryParseRevision(in ArraySegment<byte> chunk, [NotNullWhen(returnVa
+            if (buffer.Length < ObjectId.Sha1CharCount * 2)
             {
-                ParseAssert($"Log parse error, not enough data: {chunk.Count}");
+                ParseAssert($"Log parse error, not enough data: {buffer.Length}");
                 revision = default;
                 return false;
             }
 
             #region Object ID, Tree ID, Parent IDs
 
-            ReadOnlySpan<byte> array = chunk.AsSpan();
-
             // The first 40 bytes are the revision ID and the tree ID back to back
-            if (!ObjectId.TryParse(array[..ObjectId.Sha1CharCount], out ObjectId? objectId) ||
-                !ObjectId.TryParse(array.Slice(ObjectId.Sha1CharCount, ObjectId.Sha1CharCount), out ObjectId? treeId))
+            if (!ObjectId.TryParse(buffer.Slice(0, ObjectId.Sha1CharCount), out ObjectId? objectId) ||
+                !ObjectId.TryParse(buffer.Slice(ObjectId.Sha1CharCount, ObjectId.Sha1CharCount), out ObjectId? treeId))
             {
-                ParseAssert($"Log parse error, object id: {chunk.Count}({array[..ObjectId.Sha1CharCount].ToString()}");
+                ParseAssert($"Log parse error, object id: {buffer.Length}({buffer.Slice(0, ObjectId.Sha1CharCount).ToString()}");
                 revision = default;
                 return false;
             }
 
             int offset = ObjectId.Sha1CharCount * 2;
 
             // Next we have zero or more parent IDs separated by ' ' and terminated by '\n'
-            int noParents = CountParents(in array, offset);
-            if (noParents < 0)
-            {
-                ParseAssert($"Log parse error, {noParents} no of parents for {objectId}");
-                revision = default;
-                return false;
-            }
+            int noParents = CountParents(in buffer, offset);
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             int CountParents(in ReadOnlySpan<byte> array, int baseOffset)
             {
                 int count = 0;
-
                 while (baseOffset < array.Length && array[baseOffset] != '\n')
                 {
-                    // Parse error, not using ParseAssert (or increasing _noOfParseError)
-                    Debug.Assert(count == 0 || array[baseOffset] == ' ', $"Log parse error, unexpected contents in the parent array: {array[baseOffset]}/{count} for {objectId}");
-                    baseOffset += ObjectId.Sha1CharCount;
                     if (count > 0)
                     {
                         // Except for the first parent, advance after the space
                         baseOffset++;
                     }
 
+                    baseOffset += ObjectId.Sha1CharCount;
                     count++;
-                }
 
-                if (baseOffset >= array.Length || array[baseOffset] != '\n')
-                {
-                    return -1;
+                    if (baseOffset >= array.Length || !(array[baseOffset] is (byte)'\n' or (byte)' '))
```

Comment:
Does this work?
```suggestion
                    if (baseOffset >= array.Length || (array[baseOffset] is not (byte)'\n' or (byte)' '))
```

Reply:
the normal rules of negating expression, so it wold be

```
if (baseOffset >= array.Length || (array[baseOffset] is not (byte)'\n' and not (byte)' '))
```

In this case I prefer the grouping with the current

Reply:
```suggestion
                    if (baseOffset >= array.Length || ((char)array[baseOffset] is not ('\n' or ' ')))
```
seems to work